visualisation/nzvm.py

```diff
@@ -44,40 +45,72 @@ def display_data_matplot(data, max_color, hh, subplot, title, start_lat, start_l
-            subplot.set_xlabel("Longitude")
-            plot = subplot.imshow(data, interpolation='none', extent=[start_lon, end_lon, data.shape[0] * hh, 0], aspect='auto')
+            if new_subplot:
+                subplot.set_xlabel("Longitude")
+            plot = subplot.imshow(data, interpolation='nearest', cmap='jet', extent=[start_lon, end_lon, data.shape[0] * hh, 0], aspect='auto')
         else:
-            subplot.set_xlabel("Latitude")
-            plot = subplot.imshow(data, interpolation='none', extent=[start_lat, end_lat, data.shape[0] * hh, 0], aspect='auto')
-
-    plot.set_cmap("jet")
-
-    subplot.set_title(title, pad=15)        
-
+            if new_subplot:
+                subplot.set_xlabel("Latitude")
+            plot = subplot.imshow(data, interpolation='nearest', cmap='jet', extent=[start_lat, end_lat, data.shape[0] * hh, 0], aspect='auto')
+    
     if max_color != -1:
         plot.set_clim(0, max_color)
     else:
         plot.set_clim(0, int(math.ceil(np.amax(data)/1000.0)*1000.0)) #round up to nearest 1000
+            
+    if new_subplot:
     
-    legend = plt.colorbar(plot, ax=subplot)
-    legend.ax.get_yaxis().labelpad = 15
-    legend.ax.set_ylabel("Wave velocity (m/sec)", rotation=270)
+        subplot.set_title(title, pad=15)  
+        legend = plt.colorbar(plot, ax=subplot)
+        legend.ax.get_yaxis().labelpad = 15
+        legend.ax.set_ylabel("Wave velocity (m/sec)", rotation=270)
+
+    return [plot]
+
+def show_frame(t, datas, subplots, titles, max_color, hh, start_lat, start_lon, end_lat, end_lon, axis, rotation):
+    """
+    Shows a frame involving multiple images (ie cross section and perturbation
+    t: frame number to draw
+    datas: array of arrays of numpy 2d arrays. outermost array is a list of different images to plot in parallel,
+            next layer is each frame in order of time, next layer is the numpy 2d array to display
+    subplots: array of subplot instances to plot on
+    titles: array of titles for each subplot
+    max_color: maximum velocity (m/s) the chart should go up to (-1 for auto)
+    hh: cell size (resolution) of VM (metres) (or 0 for no size)
+    start_lat: latitude of top left point of cross section (or 361 for no latitude)
+    start_lon: longitude of top left point of cross section (or 361 for no longitude)
+    end_lat: latitude of bottom right point of cross section (or 361 for no latitude)
+    end_lon: longitude of bottom right point of cross section (or 361 for no longitude)
+    axis: 'x' or 'y' for which axis the cross section is on
+    rotation: degrees of rotation the model is, where 0 means the x axis follows the same latitude
+    """
+
+    blit_array = []
+    for i in range(len(datas)):
+        blit_array += display_data_matplot(datas[i][t], max_color, hh, subplots[i], titles[i], start_lat, start_lon, end_lat, end_lon, axis, rotation)
+
+    return blit_array
 
 def main():
```

Comment:
Just as a note:

This function is now quite complex. Aka doing too much / isn't straightforward to follow.

In future you should probably split off a few of the cases into separate functions so that the logic is easier to follow. Potentially using support functions to do common logic rather than a large function with lots of conditional logic for two different workflows.